src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -572,6 +572,25 @@ public function text()
     }
 
     /**
+     * Returns the flattened node value of the first node of the list.
+     *
+     * Surrounding spaces/newlines are removed,
+     * and consecutive spaces/newlines are reduced to one space.
+     *
+     * @return string The flattened node value
+     *
+     * @throws \InvalidArgumentException When current node is empty
+     */
+    public function flatText()
```

Comment:
I don't like the proposed function name (`flatText()`) because this is usually a verb ("flatten") and here we're proposing to use an adjective ("flat"). I've been looking for in the standard libraries of some programming languages (Go, JavaScript, Java, C#, ...) but nobody defines a function like that (they only implement something like our trim(), ltrim(), rtrim(), etc.) Even Laravel doesn't define anything like this in Illuminate\Str.

Maybe `trimAll()` would be a better name? Or `flatten()`? Or `trimText()`, `flattenText()`? I'm not sure.